### PR TITLE
refactor: take oracle data

### PR DESCRIPTION
**Pull request by 0xsambugs**
@0xged mentioned that since our new oracles can take some bytes as part of the quote, we could send those bytes as part of the swap

This PR adds support for this

### CODE REVIEW

File: contracts/DCAHub/DCAHubSwapHandler.sol [hunk 1]
```diff
@@ -150,31 +166,26 @@ abstract contract DCAHubSwapHandler is ReentrancyGuard, DCAHubConfigHandler, IDC
       _pairInSwap.tokenA = _tokens[indexTokenA];
       _pairInSwap.tokenB = _tokens[indexTokenB];
 
-      uint256 _amountToSwapTokenA;
-      uint256 _amountToSwapTokenB;
-
-      (_amountToSwapTokenA, _amountToSwapTokenB, _pairInSwap.intervalsInSwap) = _getTotalAmountsToSwap(
+      (_pairInSwap.totalAmountToSwapTokenA, _pairInSwap.totalAmountToSwapTokenB, _pairInSwap.intervalsInSwap) = _getTotalAmountsToSwap(
```
Comment (0xsambugs) [inline contracts/DCAHub/DCAHubSwapHandler.sol:169]: Will test that `totalAmountToSwapTokenA` and `totalAmountToSwapTokenB` are expose correctly in another PR

File: contracts/DCAHub/DCAHubSwapHandler.sol [hunk 2]
```diff
@@ -150,31 +166,26 @@ abstract contract DCAHubSwapHandler is ReentrancyGuard, DCAHubConfigHandler, IDC
...
-      _total[indexTokenB] += _amountToSwapTokenB;
-
-      uint256 _magnitudeA = tokenMagnitude[_pairInSwap.tokenA];
-      uint256 _magnitudeB = tokenMagnitude[_pairInSwap.tokenB];
+      _total[indexTokenA] += _pairInSwap.totalAmountToSwapTokenA;
+      _total[indexTokenB] += _pairInSwap.totalAmountToSwapTokenB;
 
-      (_pairInSwap.ratioAToB, _pairInSwap.ratioBToA) = _calculateRatio(
+      // Note: it would be better to calculate the magnitudes here instead of inside `_calculateRatio`, but it throws a "stack to deep" error
+      PairMagnitudes memory _magnitudes;
```
[the remark this answers is not in the document]
Comment (0xsambugs) [reply contracts/DCAHub/DCAHubSwapHandler.sol:179]: Yes 😭

File: contracts/interfaces/IDCAHub.sol [hunk 3]
```diff
@@ -426,6 +426,10 @@ interface IDCAHubSwapHandler {
     address tokenA;
     // The address of the other token
     address tokenB;
+    // The total amount of token A swapped in this pair
+    uint256 totalAmountToSwapTokenA;
```
[the remark this answers is not in the document]
Comment (0xged) [reply contracts/interfaces/IDCAHub.sol:430]: Yeah, we now can know the volume per pair! 🥳

File: test/unit/DCAHub/dca-hub-swap-handler.spec.ts [hunk 4]
```diff
@@ -452,21 +455,23 @@ contract('DCAHubSwapHandler', () => {
...
         expect(ratioAToB).to.equal(tokenA.magnitude.mul(tokenB.magnitude).div(expectedRatioBToA));
         expect(ratioBToA).to.equal(expectedRatioBToA);
       });
+      then('magnitudes are calculated correctly', () => {
+        expect(magnitudes.magnitudeA).to.equal(tokenA.magnitude);
+        expect(magnitudes.magnitudeB).to.equal(tokenB.magnitude);
+      });
+      then('oracle is called correctly', () => {
+        expect(priceOracle.quote).to.have.been.calledOnceWith(tokenB.address, tokenB.magnitude, tokenA.address, BYTES);
+      });
```
Comment (0xged) [inline test/unit/DCAHub/dca-hub-swap-handler.spec.ts:474]: 👏

**Review by 0xged — APPROVED** [review]
Nice ❤️  - sucks that `viaIR` to avoid `stack too deep` is so new, kinda dangerous to use it rn.  Grea job on this Sam!

**Pull request merged**